src/main/java/appeng/thirdparty/codechicken/lib/model/pipeline/BakedPipeline.java

```diff
@@ -0,0 +1,434 @@
+	}
+
+	/**
+	 * Used to create a BakedPipeline.
+	 *
+	 * @return The builder.
+	 */
+	public static Builder builder()
+	{
+		return new Builder();
+	}
+
+	/**
+	 * Used to reset the pipeline for the next quad.
+	 * MUST be called between quads.
+	 *
+	 * @param format The format.
+	 */
+	public void reset( VertexFormat format )
+	{
+		reset( CachedFormat.lookup( format ) );
+	}
+
+	/**
+	 * Used to reset the pipeline for the next quad.
+	 * MUST be called between quads.
+	 *
+	 * @param format The format.
+	 */
+	public void reset( CachedFormat format )
+	{
+		unpacker.reset( format );
+		for( PipelineElement element : elements )
+		{
+			element.reset( format );
+		}
+		first = null;
+	}
+
+	/**
+	 * Get an element from the pipeline.
+	 *
+	 * @param name The name of the element.
+	 * @param clazz The Class of the element, used to safe cast.
+	 *
+	 * @return The element.
+	 */
+	public <T extends IPipelineConsumer> T getElement( String name, Class<T> clazz )
+	{
+		PipelineElement element = nameLookup.get( name );
+		if( element != null )
+		{
+			if( !clazz.isAssignableFrom( element.consumer.getClass() ) )
+			{
+				throw new IllegalArgumentException( "Element with name " + name + " is not assignable from reference class." );
+			}
+			return clazz.cast( element.consumer );
+		}
+		throw new IllegalArgumentException( "Element with name " + name + " does not exist." );
+	}
+
+	/**
+	 * Used to enable an element on the pipeline with the specified name.
+	 *
+	 * @param name The elements name.
+	 */
+	public void enableElement( String name )
+	{
+		setElementState( name, true );
+	}
+
+	/**
+	 * Used to disable an element on the pipeline with the specified name.
+	 *
+	 * @param name The elements name.
+	 */
+	public void disableElement( String name )
+	{
+		setElementState( name, false );
+	}
+
+	/**
+	 * Used to set the state of an element on the pipeline.
+	 *
+	 * @param name The name of the element.
+	 * @param enabled The state to set it to.
+	 */
+	public void setElementState( String name, boolean enabled )
+	{
+		PipelineElement element = nameLookup.get( name );
+		if( element != null )
+		{
+			element.isEnabled = enabled;
+			return;
+		}
+		throw new IllegalArgumentException( "Element with name " + name + " does not exist." );
+	}
+
+	/**
+	 * Call when you are ready to use the pipeline.
+	 * This builds the internal state of the Elements getting things ready to transform.
+	 *
+	 * @param collector The IVertexConsumer that should collect the transformed quad.
+	 */
+	public void prepare( IVertexConsumer collector )
+	{
+		IPipelineConsumer next = null;
+		for( PipelineElement element : elements )
+		{
+			if( element.isEnabled )
+			{
+				if( first == null )
+				{
+					first = element.consumer;
+				}
+				else
+				{
+					next.setParent( element.consumer );
+				}
+				next = element.consumer;
+			}
+		}
+		next.setParent( collector );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "next" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS2259)